tap_postgres/sync_strategies/logical_replication.py

```diff
@@ -217,14 +217,90 @@ def row_to_singer_message(stream, row, version, columns, time_extracted, md_map,
         version=version,
         time_extracted=time_extracted)
 
-def consume_message(streams, state, msg, time_extracted, conn_info, end_lsn):
-    payload = json.loads(msg.payload)
-    lsn = msg.data_start
-
-    streams_lookup = {}
-    for s in streams:
-        streams_lookup[s['tap_stream_id']] = s
-
+def consume_message_format_2(payload, conn_info, streams_lookup, state, time_extracted, lsn):
+    ## Action Types:
+    # I = Insert
+    # U = Update
+    # D = Delete
+    # B = Begin Transaction
+    # C = Commit Transaction
+    # M = Message
+    # T = Truncate
+    record_message = None
+
+    action = payload['action']
+    if action not in ['U', 'I', 'D']:
+        LOGGER.debug("Skipping message of type %s", action)
+        yield None
+    else:
+        tap_stream_id = post_db.compute_tap_stream_id(conn_info['dbname'], payload['schema'], payload['table'])
+        if streams_lookup.get(tap_stream_id) is None:
+            yield None
+        else:
+            target_stream = streams_lookup[tap_stream_id]
+            stream_version = get_stream_version(target_stream['tap_stream_id'], state)
+            stream_md_map = metadata.to_map(target_stream['metadata'])
+
+            desired_columns = [col for col in target_stream['schema']['properties'].keys() if sync_common.should_sync_column(stream_md_map, col)]
+
+            if payload['action'] == 'I':
+                col_names = []
+                col_vals = []
+
+                for column in payload['columns']:
+                    if column['name'] in set(desired_columns):
+                        col_names.append(column['name'])
+                        col_vals.append(column['value'])
+
+                col_names = col_names + ['_sdc_deleted_at']
+                col_vals = col_vals + [None]
+                if conn_info.get('debug_lsn'):
+                    col_names = col_names + ['_sdc_lsn']
+                    col_vals = col_vals + [str(lsn)]
+                record_message = row_to_singer_message(target_stream, col_vals, stream_version, col_names, time_extracted, stream_md_map, conn_info)
+
+            elif payload['action'] == 'U':
+                col_names = []
+                col_vals = []
+                for column in payload['columns']:
+                    if column['name'] in set(desired_columns):
+                        col_names.append(column['name'])
+                        col_vals.append(column['value'])
+
+                col_names = col_names + ['_sdc_deleted_at']
+                col_vals = col_vals + [None]
+
+                if conn_info.get('debug_lsn'):
+                    col_vals = col_vals + [str(lsn)]
+                    col_names = col_names + ['_sdc_lsn']
+                record_message = row_to_singer_message(target_stream, col_vals, stream_version, col_names, time_extracted, stream_md_map, conn_info)
```

Comment:
They do, but we wanted to keep them separate since they're separate events. If there are unknown cases where a PG instance makes Updates different, this will need to change.

Reply:
I would really rather combine them if they're the same. It would be very easy to split the cases in the future if we eventually do want to handle them differently, but right now they do the same thing and I think our branching should reflect that.

Reply:
That's fair. I'll merge them.